### PR TITLE
refactor: move GA libraries source of truth

### PR DESCRIPTION
**Pull request by coryan**
The CMake files will need to know the list of GA libraries to provide a single "build all" configuration parameter.  We already have infrastructure to export variables from CMake to Bazel, it makes sense to move these there too.

Part of the work for #9333

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10603)

### CODE REVIEW

File: cmake/CreateBazelConfig.cmake [hunk 1]
```diff
@@ -115,13 +115,14 @@ function (export_list_to_bazel filename VAR)
         APPEND "${filename}"
         [=[
 """Automatically generated unit tests list - DO NOT EDIT."""
```
Comment (dbolduc) [inline cmake/CreateBazelConfig.cmake:117]: Probably not in this PR, but: `s/unit tests list/list(s)/` seems more appropriate

Comment (devbww) [inline cmake/CreateBazelConfig.cmake:117]: Nit: Perhaps the "unit tests" string should be an `export_list_to_bazel` parameter.

File: doc/contributor/howto-guide-adding-generated-libraries.md [hunk 2]
```diff
@@ -172,48 +172,11 @@ knows about one such subdirectory. You may need to manually update the
...
- EXPERIMENTAL_LIBRARIES = [
-+    # Introduced in 2022-05
-+    "bms",
-     # Introduced in 2022-04
-     "dataplex",
-     "dialogflow_cx",
-```
-
-</details>
+Manually edit the `google/GoogleCloudCppFeatures.cmake` file to include the new
```
Comment (dbolduc) [inline doc/contributor/howto-guide-adding-generated-libraries.md:175]: "Manually edit `cmake/GoogleCloudCppFeatures.cmake` to include the new ..."

**Review by devbww — APPROVED** [review]

File: doc/contributor/howto-guide-adding-generated-libraries.md [hunk 3]
```diff
@@ -300,7 +263,7 @@ index c4ce00489..1858b48dc 100755
 
 ```shell
 git commit -m"Manually update READMEs, quickstart, and top-level stuff" \
-   "google/cloud/${library}" BUILD.bazel CHANGELOG.md ci README.md
+   "google/cloud/${library}" cmake CHANGELOG.md ci README.md
```
Comment (dbolduc) [inline doc/contributor/howto-guide-adding-generated-libraries.md:266]: lexicographical supernit: CHANGELOG.md < ci < cmake

File: cmake/GoogleCloudCppFeatures.cmake [hunk 4]
```diff
@@ -15,13 +15,121 @@
...
+    # Introduced circa 2022-08-17
+    "apikeys"
+    # This is WIP, it needs a number of hand-crafted APIs.
+    "pubsublite")
+
+set(GOOGLE_CLOUD_CPP_TRANSITION_FEATURES # cmake-format: sorted
+    # Promoted to GA circa 2022-10-12
+    "batch")
+
+set(GOOGLE_CLOUD_CPP_GA_FEATURES
```
Comment (dbolduc) [inline cmake/GoogleCloudCppFeatures.cmake:37]: are you sure you want to call these things `FEATURES` and not `LIBRARIES` ?

I think I would expect to see `storage-grpc`, `grafeas`, `generator`, `opentelemetry` also listed as features.

File: cmake/CreateBazelConfig.cmake (hunk 1 above)
Comment (coryan) [reply cmake/CreateBazelConfig.cmake:117]: I will fix that in a follow-up PR, it changes dozens of files unrelated to the current problem.

Comment (coryan) [reply cmake/CreateBazelConfig.cmake:117]: Ack.

File: cmake/GoogleCloudCppFeatures.cmake (hunk 4 above)
Comment (coryan) [reply cmake/GoogleCloudCppFeatures.cmake]: Fixed.

File: doc/contributor/howto-guide-adding-generated-libraries.md (hunk 3 above)
Comment (coryan) [reply doc/contributor/howto-guide-adding-generated-libraries.md]: Done.

File: doc/contributor/howto-guide-adding-generated-libraries.md (hunk 2 above)
Comment (coryan) [reply doc/contributor/howto-guide-adding-generated-libraries.md]: Done.

File: doc/contributor/howto-guide-adding-generated-libraries.md [hunk 5]
```diff
@@ -172,48 +172,11 @@ knows about one such subdirectory. You may need to manually update the
...
- EXPERIMENTAL_LIBRARIES = [
-+    # Introduced in 2022-05
-+    "bms",
-     # Introduced in 2022-04
-     "dataplex",
-     "dialogflow_cx",
-```
-
-</details>
+`GOOGLE_CLOUD_CPP_EXPERIMENTAL_FEATURES` and note in a comment when the library
```
Comment (devbww) [inline doc/contributor/howto-guide-adding-generated-libraries.md:178]: s/FEATURES/LIBRARIES/

File: doc/contributor/howto-guide-adding-generated-libraries.md [hunk 6]
```diff
@@ -172,11 +172,11 @@ knows about one such subdirectory. You may need to manually update the
 
 ## Update the root files
 
-Manually edit the `google/GoogleCloudCppFeatures.cmake` file to include the new
-target. If you are generating a GA library, add it to
-`GOOGLE_CLOUD_CPP_GA_FEATURES`. Otherwise, if you are generating an experimental
-library, add it to `GOOGLE_CLOUD_CPP_EXPERIMENTAL_FEATURES` and note in a
-comment when the library was generated.
+Manually edit `google/GoogleCloudCppFeatures.cmake` to include the new target.
```
Comment (dbolduc) [inline doc/contributor/howto-guide-adding-generated-libraries.md:175]: s/google/cmake/

Comment (coryan) [reply doc/contributor/howto-guide-adding-generated-libraries.md]: Fixed.

File: doc/contributor/howto-guide-adding-generated-libraries.md (hunk 5 above)
Comment (coryan) [reply doc/contributor/howto-guide-adding-generated-libraries.md]: Fixed.

**Review by dbolduc — APPROVED** [review]

**Pull request merged**